### ISSUE TITLE
Dispose doesnt work when page is refreshed (server-side blazor)

### ISSUE THREAD

**Issue by FeeSaver** [aspnet/AspNetCore#5633]
Hello I was trying to make the fetchdata example update values without refreshing page or clicking buttons. And I found out that I can use timer which seemed like working ok. But after testing with different data it was acting strange and I found out that page doesnt dispose the timer when I go to different page and its not displayed anymore. 
That was solved by adding dispose but it didnt solve different issue.. When I press F5 while on /fetchdata page it reloads and doesnt dispose the timer. And then it runs twice. I tried it by adding int into ForecastService and it increment everytime GetForecastAsync is called. After 10 refreshes it increment +10 every second instead of +1. 
I hope it makes sense. 

```
@functions {
WeatherForecast[] forecasts;
Timer timer;
private async void Update()
{
    forecasts = await ForecastService.GetForecastAsync(DateTime.Now);
    StateHasChanged();
}

protected override async Task OnInitAsync()
{
    timer = new Timer(1000);
    timer.Elapsed += (sender, args) => Update();
    timer.Start();
    Update();
}

public void Dispose()
{
    timer.Dispose();
}
}
```

**Comment by rynowak** [comment]
Clearing milestone, backlog doesn't seem right for this.

**Comment by javiercn** [comment]
@SteveSandersonMS Should we just attach ourselves to `beforeunload` to dispose the connection / orderly tear down the circuit?

**Comment by SteveSandersonMS** [comment]
I thought we had resolved this already. I think @pranavkm implemented it.

But if this is still reproducible, we need to be sure that Dispose to be called in all cases, not only when the beforeunload event runs (it wouldn’t if the user just closed their browser).

**Comment by javiercn** [comment]
@SteveSandersonMS Yeah, the circuit gets set to disconnected when the connection is lost and eventually gets disposed. We can probably close this then. Should we try and be more proactive when we can?

**Comment by javiercn** [comment]
Closing it as it is already done. Keep in mind that dispose will fire up only after the circuit gets reclaimed and not immediately after refresh.

**Comment by SteveSandersonMS** [comment]
I agree we should consider evicting immediately if we get notification from the client that they are unloading the document. Could we have a backlog item for that?

**Comment by javiercn** [comment]
I added that to https://github.com/aspnet/AspNetCore/issues/6358
